ipatests/test_integration/test_installation.py

```diff
@@ -254,6 +254,85 @@ def test_replica_ca_install_with_skip_schema_check(self):
         tasks.install_replica(self.master, self.replicas[1], setup_ca=False)
         tasks.install_ca(self.replicas[1], extra_args=["--skip-schema-check"])
 
+    def test_certmonger_reads_token_HSM(self):
+        """Test if certmonger reads the token in HSM
+
+        This is to ensure added HSM support for FreeIPA. This test adds
+        certificate with sofhsm token and checks if certmonger is tracking
+        it.
+
+        related : https://pagure.io/certmonger/issue/125
+        """
+        test_service='test/%s'%self.master.hostname
+        pkcs_passwd='Secret123'
+        pin='123456'
+        noise_contents=("sjdhkjhdkjahdkjsahdhkjsahdkjsahdkjHDJSADHSADASDJHAA"
+                        "ADAJSDKJSAHDJSAHDJASHDJHSADKJAHSDKJAHDSKJAHSDKJHASA"
+                        "ASJDHASJDHASJDHJASHDJASHDKJASHDJSAHDJSAHDJSAHDJSWSH")
+        noise_file='/tmp/noise'
+        self.master.master.put_file_contents(noise_file, noise_contents)
```

Comment:
test_ipapython/test_certdb.py has an example of a better way to generate a noise file.